resources/intro-to-observables.md

```diff
@@ -0,0 +1,127 @@
+
+We also encounter this with DOM events, timer intervals, web workers and web sockets.
+
+**Don't we already have a few ways to handle asynchronous data?** 
+Yes, we can use callback functions and promises for example. But Observables are slightly different as they allow us to process **streams of data**.
+
+## Why should I care?
+
+Observables play a key part in something called **reactive programming**. This is the concept of programming with asynchronous data streams (a sequence of values in time). You can choose to listen to these streams and, as data comes through them, react accordingly. 
+
+Data streams are functional - so you can use streams as inputs into other streams, combine them, filter them, etc using a variety of helpful functions. 
+
+Angular has Observables baked into core parts of its library, such as its event system and HTTP client service, using a third-party library called RxJS (Reactive Extensions). In addition, we're starting to see more React projects use RxJS to take advantage of Observables to manage streams.
+
+## What is an Observable?
+
+Observables originate from something called the **observer design pattern**. 
+
+> The observer pattern is a software design pattern in which an object, called the **subject**, maintains a list of its dependents, called **observers**, and notifies them automatically of any state changes, usually by calling one of their methods. ([source](https://en.wikipedia.org/wiki/Observer_pattern)) 
+
+Ok, so that definition is pretty wordy and technical. 
+
+In simple terms, the pattern involves an object (called an observable) that will share (or emit) data as it arrives over time. Throughout your code, many different objects (called observers) can sign up to be notified when that data arrives (called subscribing). So, whenever our observable recieves its data, it will broadcast that information out and the observers will be listening for it.  
+
+Some key points to note about the pattern:
+
+- You can have one observable and many observers (a one-to-many dependancy). That is, one broadcaster and an unlimited number of recipients.
+- When that observable changes, all observers should be updated/notified immediately and automatically
+- It is the observer's responsibility to register and unregister themselves from the observable (to be notified of changes) and to update their own state when notified.
+
+## So... an Observable?
+
+Think of an observable as a box that you create to hold data from a service or function whose data or items will arrive over time. 
+
+As the data arrives, the box notifies anyone that has requested they be updated and it does so in the way they've asked. (Note: once new data arrives, the previous data no longer exists in the box. It's up to the listener to maintain the previous data if needed.)  
```

Comment:
> (Note: once new data arrives, the previous data no longer exists in the box. It's up to the listener to maintain the previous data if needed.)  

This might become confusing when/if someone reading this discovers replay subjects!

Reply:
@AbdellaToronto good point, I'll remove this note